<launch>
  <include file="$(find topower_v1_moveit)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <rosparam file="$(find topower_v1)/config/topower_v1_control.yaml" command="load"/>

  <!-- controller manager & hardware interface -->
  <node name="topower_v1_hw" pkg="topower_v1" type="topower_v1_hw" output="screen" ns="topower_v1">
  </node>

  <!-- ros controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
	output="screen" ns="topower_v1" args="joint_state_controller
	  wheel_l_effort_controller wheel_r_effort_controller
    cam_pan_position_controller cam_tilt_position_controller arm_controller
    gripper_l_position_controller"/>

  <!-- for moveit arm control -->
  <include file="$(find topower_v1_moveit)/launch/move_group.launch">
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="fake_execution" value="false"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="false"/>
  </include>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
  respawn="false" output="screen">
    <remap from="/joint_states" to="/topower_v1/joint_states"/>
  </node>

  <node name="arm_control_node" pkg="topower_v1" type="arm_control_node.py" output="screen">
    <remap from="/joint_states" to="/topower_v1/joint_states"/>
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find topower_v1)/launch/topower_v1_moveit.rviz"/>


</launch>